plugins/catalog-backend/src/service/NextEntitiesCatalog.test.ts

```diff
@@ -32,6 +33,14 @@ describe('NextEntitiesCatalog', () => {
     ids: ['POSTGRES_13', 'POSTGRES_9', 'SQLITE_3'],
   });
 
+  const permissionClient = {
```

Comment:
I think this can be typed as `PermissionClient` to avoid all those `as any` annotations below

Reply:
yeah I had tried that initially but it'd need more work to get the types happy and to actually have a working mock. there's a TODO here for just that:
```
// TODO(authorization-framework: pass mock permission client, extend test suite to check authz behavior)
```
wasn't sure whether to bite the bullet and include that as part of this change or handle that separately.

Reply:
Oh weird that it wouldn't pass type checking, but I know typing jest mocks can be tricky. As for extending the tests to check authz behavior - seems okay to do that any time before these changes get merged into master

Reply:
yeah it was complaining about all of `permissionClient`'s missing private methods and such. this was really just a way to get the tests not to fail when something tries to call `permissionApi.authorize`, but don't think it can be the final solution

Reply:
Interesting, I didn't realize typescript checked for private methods/variables! Perhaps an interface for the `PermissionClient` could help here, but not super important right now. Another approach could involve making a real `PermissionClient` instance with `permission.enabled` set to `false`. Either way we can improve the tests in the upcoming PRs for catalog integration

Reply:
yeah honestly i'm pretty surprised about that as well. might play around with it a bit just to figure out what's going on, but then leave the implementation for a separate time